core/src/nodes/node.rs

```diff
@@ -347,32 +346,36 @@ where
     }
 }
 
-// TODO:
-/*impl<TTrans> fmt::Debug for NodeEvent<TTrans>
-where TTrans: Transport,
-      <TTrans::Listener as Stream>::Error: fmt::Debug,
+impl<TMuxer, TUserData> fmt::Debug for NodeEvent<TMuxer, TUserData>
+where 
+    TMuxer: muxing::StreamMuxer,
+    TUserData: fmt::Debug
 {
-    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
-            NodeEvent::Incoming { ref listen_addr, .. } => {
-                f.debug_struct("NodeEvent::Incoming")
-                    .field("listen_addr", listen_addr)
+            NodeEvent::InboundSubstream { .. } => {
+                f.debug_struct("NodeEvent::OutboundClosed")
+                    .field("substream", &"...")
```

Comment:
Would it not be useful to include the substream debug output here? After all, this was the primary motivation for adding the `Debug` impl for `SubstreamRef` IIRC.